src/controller/c/supervisor.c

```diff
@@ -3675,3 +3918,157 @@ const char *wb_supervisor_field_get_type_name(WbFieldRef field) {
       return "";
   }
 }
+
+const char *wb_supervisor_proto_get_type_name(WbProtoRef proto) {
```

Comment:
The following functions don't seem to call `robot_check_supervisor`: `wb_supervisor_proto_get_type_name`, `wb_supervisor_proto_is_derived`, `wb_supervisor_proto_get_number_of_fields`. Is that intentional?

Reply:
For that, I was mirroring the `wb_supervisor_field_get_*` functions, which do not call `robot_check_supervisor` for values which can be obtained from the struct.